js/app/Visualization/UI/BasicSidebar.js

```diff
@@ -176,8 +176,39 @@ define([
             self.node.find(".vessel_id .flag").html("---");
           }
 
-          self.node.find(".vessel_id .imo").html(data.imo || "---");
-          self.node.find(".vessel_id .mmsi").html(data.mmsi || "---");
+          if (data.imo) {
```

Comment:
If you sum up little code that wouldn't make difference you always end up having lots of code that end up on difficult code to maintain.